libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -263,7 +272,26 @@ AP_OSD::AP_OSD()
 
 void AP_OSD::init()
 {
-    switch (osd_types(osd_type.get())) {
+    const AP_OSD::osd_types types[OSD_MAX_INSTANCES] = {osd_types(osd_type.get()), osd_types(osd_type2.get())};
+    for (uint8_t instance = 0; instance < OSD_MAX_INSTANCES; instance++) {
+        if (init_backend(types[instance], instance)) {
+            if (instance == 0) {
+                hal.scheduler->thread_create(FUNCTOR_BIND_MEMBER(&AP_OSD::osd_thread, void), "OSD", 1280, AP_HAL::Scheduler::PRIORITY_IO, 1);
```

Comment:
this means we don't create thread if 1st instance fails to init, add thread of _backend_count is > 0

Reply:
auch, you're right